config/plugins/visualizations/annotate_image/src/script.js

```diff
@@ -437,11 +437,13 @@ window.bundleEntries.load = function (opt) {
         });
     };
 
+    const slash_cleanup = /(\/)+/g;
+    const safe_download_url = `${options.root}/${dataset.download_url}`.replace(slash_cleanup, "/");
     $.ajax({
-        url: dataset.download_url,
+        url: safe_download_url,
```

Comment:
This should probably be fixed in the backend, such that the correct url is provided through the api. It seems to be serialized here: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/managers/hdas.py#L450.

Reply:
I thought about it. But when I looked into code base and previous PRs, I have noticed that the Galaxy prefix is added on the client side mostly. I am quite new to Galaxy project so I do not know all the architectural decisions regarding API interfaces. Moreover `url_for` will build the URL based on route and I am not quite sure where the prefix is exposed to prepend the URL returned by `url_for`.

Reply:
Download url is relative, so if you don't globally set the prefix on app load (we should 100% do that) you'll have to add the app prefix.

Reply:
Ok, yes, this is how it is being handled right now. Thank you for fixing this bug!